server/src/main/java/io/crate/expression/operator/any/AnyLikeOperator.java

```diff
@@ -66,7 +67,16 @@ boolean matches(String probe, String candidate) {
     }
 
     @Override
-    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, @NotNull List<?> nonNullValues, Context context) {
+    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, Literal<?> candidates, Context context) {
+        if (ArrayType.dimensions(candidates.valueType()) > 1) {
+            return null;
+        }
```

Comment:
Can't we unwrap and pre-filter in this case?

Reply:
It's also okay to look into this as a follow up / feature if it didn't regress anything.

Reply:
Justed doubled checked, `like` variants are still throwing `UnsupportedException`(which is an acceptable exception) and `AnyNeq` and `AnyRange` operators used to throw `ClassCastExceptions` that is fixed by this PR. I will add more entries in the release notes.

Reply:
Working on an extra release note entry https://github.com/crate/crate/pull/16667/commits/f1ba3148b6a3756ecbfc53a5dd6b9b0dd2bc3461, please let me know if this could be too much details(since it was not user reported..)